core/src/main/scala/org/ensime/indexer/FileWatchers.scala

```diff
@@ -114,18 +114,20 @@ class ClassfileWatcher(
 
     val root = vfs.vfile(config.root)
 
-    // must remove then add to avoid leaks
-    for {
-      d <- config.targetClasspath
-      dir = vfs.vfile(d)
-      _ = fm.removeFile(dir)
-      _ = fm.addFile(dir)
-      ancestor <- ancestors(dir)
-      if ancestor.getName isAncestor root.getName
-      _ = workaround.removeFile(ancestor)
-      _ = workaround.addFile(ancestor)
-    } {
-      // side effects in the for comprehension
+    if (!config.disableClassMonitoring) {
+      // must remove then add to avoid leaks
+      for {
+        d <- config.targetClasspath
+        dir = vfs.vfile(d)
```

Comment:
it would be more canonical to put the `if` statement in the for comprehension instead of outside it (will also result in a much smaller diff)